storageClient/api.go

```diff
@@ -80,9 +85,36 @@ type PrimeResponseBody struct {
 
 type ChallengeRequest struct {
 	EmailEACPChallenge EmailEACPChallengeRequest `json:"email_eacp"`
+	TozIDEACPChallenge TozIDEACPChallengeRequest `json:"tozid_eacp"`
 }
 
 type EmailEACPChallengeRequest struct {
 	TemplateName string `json:"template_name"`
 	OTPRequest
 }
+
+// TozIDEACPChallengeRequest wraps parameters needed to activate a TozID EACP on a note.
+type TozIDEACPChallengeRequest struct {
+	NoteID        string
+	IDString      string
+	ExpirySeconds int64 `json:"expiry_seconds"`
+}
+
+// ChallengeResponse wraps parameters for all activated challenges on a note.
+type ChallengeResponse struct {
+	EmailEACPChallenge *EmailEACPChallengeResponse `json:"email_eacp,omitempty"`
+	TozIDEACPChallenge *TozIDEACPChallengeResponse `json:"tozid_eacp,omitemtpy"`
+}
+
+type EmailEACPChallengeResponse struct {
+	EmailAddress string `json:"email_address"`
+	TemplateName string `json:"template"`
+	ProviderLink string `json:"provider_link"`
+	OTP          string `json:"otp"`
+}
+
+// TozIDEACPChallengeRequest wraps parameters for an activated TozID EACP on a note.
+type TozIDEACPChallengeResponse struct {
+	ExpiresAt time.Time `json:"expires_at"`
+	OTP       string    `json:"otp"`
```

Comment:
change name to something indicating this is a challenge value that must be signed over in an identity token as the nonce value.